### PR TITLE
[Doc] Fix --user in dev install instruction.

### PR DESCRIPTION
**Pull request by ailzhang**
Thanks @yuanming-hu for trying it out!

### CODE REVIEW

File: docs/lang/articles/contribution/dev_install.md [hunk 1]
```diff
@@ -114,10 +114,10 @@ installer.
 1. Clone the Taichi repo **recursively**, and build:
 
   ```bash
-  git clone --recursive https://github.com/taichi-dev/taichi --depth=1 --branch=master
+  git clone https://github.com/taichi-dev/taichi --depth=1 --branch=master
```
Comment (yuanming-hu) [inline docs/lang/articles/contribution/dev_install.md:117]: I believe we still need to recursively clone the submodules? :-)

**Review by k-ye — APPROVED** [review]
Thanks!

File: docs/lang/articles/contribution/dev_install.md [hunk 2]
```diff
@@ -224,7 +224,7 @@ sudo systemctl restart docker
 
 - If `python3 setup.py develop/install` gives `permission denied` error, it means you're
   installing into system python without write permission. You can work around this by:
-  - `python3 setup.py --user develop/install`
+  - `python3 setup.py develop/install --user`
```
Comment (yuanming-hu) [inline docs/lang/articles/contribution/dev_install.md:227]: I tried to use this command, yet got the following error message:
```python
>>> python3 setup.py develop/install
[]
['taichi', 'taichi.misc', 'taichi.tools', 'taichi.core', 'taichi.tests', 'taichi.snode', 'taichi.lang', 'taichi.aot']
invalid command name 'develop/install'
```

It seems that `develop/install` is not a registered command?

**Review by yuanming-hu — APPROVED** [review]
LGTM! Thank you! :-)

**Pull request merged**